src/main/java/seedu/duke/Parser.java

```diff
@@ -0,0 +1,83 @@
+package seedu.duke;
+
+/**
+ * Handles the parsing of user input.
+ */
+public class Parser {
+
+    /**
+     * Parses user input and returns a command corresponding to user input.
+     *
+     * @param str String containing user input.
+     * @return Command corresponding to user input.
+     * @throws DukeException if user input is invalid or does not correspond to any existing command.
+     */
+    public static Command parse(String str) throws DukeException {
+        String full = str;
+        String[] arr = full.split(" ", 2);
+        String s = arr[0];
+        Command c = null;
+
+        validateInput(full);
+
+        if (s.equals("done")) {
+            c = new DoneCommand(Integer.valueOf(arr[1]) - 1);
+        } else if (s.equals("delete")) {
+            c = new DeleteCommand(Integer.valueOf(arr[1]) - 1);
+        } else if (s.equals("edit")) {
+            String information = arr[1].trim();
+            String[] split = information.split(" ", 2);
+            c = new EditCommand(Integer.valueOf(split[0]) - 1, split[1]);
+        } else if (s.equals("todo") || s.equals("deadline") || s.equals("event")) {
+            c = new AddCommand(s, arr[1].trim());
+        } else if (s.toLowerCase().equals("list")) {
+            c = new ListCommand();
+        } else if (s.equals("find")) {
+            c = new FindCommand(arr[1].trim());
+        } else {
+            if (s.equals("bye")) {
+                c = new ExitCommand();
+            }
+        }
+
+        return c;
+    }
+
+    /**
+     * Parses user input and detects if it corresponds to existing commands.
+     *
+     * @param input String containing user input.
+     * @throws DukeException if user input is invalid.
+     */
+    public static void validateInput(String input) throws DukeException {
```

Comment:
Lines 61 to 79 a bit messy, especially the conditionals. Maybe try to incorporate validateInput into the respective if else statements from line 23 to 40 instead?